base-permissions/src/commonMain/kotlin/Permission.kt

```diff
@@ -89,21 +90,24 @@ open class PermissionsBuilder(val context: PermissionContext = defaultPermission
         } ?: throw Error("Permission state repo factory was not registered for $permission")
 }
 
+private val defaultPermissionDispatcher by lazy {
+    singleThreadDispatcher("Permissions")
+}
+
 /**
  * Manager to request the [PermissionStateRepo] of a given [Permission]
  * @param builder The [PermissionsBuilder] to build the [PermissionManager] associated with each [Permission]
  * @param coroutineContext The [CoroutineContext] to run permission checks from
  */
 class Permissions(
     private val builder: PermissionsBuilder,
-    private val coroutineContext: CoroutineContext = singleThreadDispatcher("Permissions"),
-    private val contextCreator: CoroutineContext.(String) -> CoroutineContext = { this + singleThreadDispatcher(it) }
+    private val coroutineContext: CoroutineContext = defaultPermissionDispatcher,
```

Comment:
It was not really needed to remove the `contextCreator` (though `Provider` would be a more accurate name), you could have just changed it to return your new global dispatchers by default.

However, with new MM providing a pool will be possible. I suppose based on all discussion this is "good enough" and we can accept this, maybe change to a small pool for BT once we can, and group some other components together or something.

Reply:
IMO a `contextCreator` was a bit misleading and would create problems in a long run. a small pool would be nice indeed

Reply:
yes, hence `contextProvider` would probably be better, but I'll resolve this now.